File: python3/n-ary_tree_postorder_traversal.py
from .model import NaryNode

# Space : O(n)
# Time  : O(n)


class Solution:
    def postorder(self, root: NaryNode) -> List[int]:
        res = []

        if root:
            if root.children:
                for node in root.children:
                    res += self.postorder(node)
            res.append(root.val)

        return res

File: golang/n-ary_tree_postorder_traversal.go
package main

import (
	"golang/model"
)

// Space : O(n)
// Time  : O(n)

func naryPostorder(root *model.NaryNode) []int {
	res := []int{}

	if root != nil {
		if len(root.Children) > 0 {
			for _, node := range root.Children {
				res = append(res, naryPostorder(node)...)
			}
		}
		res = append(res, root.Val)
	}

	return res
}

